tests/test_connection_string.py

```diff
@@ -17,3 +17,24 @@ def test_connection_strings(connection_string: str):
     )
     res = cf.make_connection_params(connection_string)
     assert res["url"] == connection_string
+
+
+@pytest.mark.parametrize(
+    "connection_string",
+    [
+        "unix://tmp/foo.bar?db=1",
+        "redis://localhost/2",
+        "rediss://localhost:3333?db=2",
+    ],
+)
+def test_connection_strings_with_conn_factory_in_opts(connection_string: str):
+    sentinel_conn_factory = "django_redis.pool.SentinelConnectionFactory"
+    cf = pool.get_connection_factory(
+        path=None,
+        options={
+            "CONNECTION_FACTORY": sentinel_conn_factory,
+            "SENTINELS": [("127.0.0.1", "26739")],
+        },
+    )
+    res = cf.make_connection_params(connection_string)
```

Comment:
why check connection params?

why not check if CF is an instance of `SentinelConnectionFactory`? I would even use multiple connection factories from `parametrize` and check that the same instance is created